repository/repo.go

```diff
@@ -185,4 +189,18 @@ type Repo interface {
 	// we merely ensure that their history graph includes every commit that we
 	// intend to keep.
 	PullNotesAndArchive(remote, notesRefPattern, archiveRefPattern string) error
+
+	// MergeNotesAndArchive merges the notes and archives from `remote` into
+	// the local branch.
+	MergeNotesAndArchive(remote, notesRefPattern,
```

Comment:
I wonder if we should split this in to two separate methods; `MergeNotes` and `MergeArchives`. The reason is that we will soon want to add something like `MergeForks`, and maybe even `MergeKeys`, and we don't want to have to combine all of those together into a single method name.